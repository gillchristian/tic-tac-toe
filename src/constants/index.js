export const O = '●'
export const X = '✖'
export const s = ''

export const RESULTS = [
  /1..1..1../,
  /.1..1..1./,
  /..1..1..1/,
  /111....../,
  /...111.../,
  /......111/,
  /1...1...1/,
  /..1.1.1../
]

