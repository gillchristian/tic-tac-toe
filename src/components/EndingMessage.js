import R from 'ramda'
import React from 'react'
import PropTypes from 'prop-types'
import glamorous from 'glamorous'

import { isO, isX, isMark } from '../utils/game'
import { X, O } from '../constants'

const Span = glamorous.span({
  fontFamily: '"Gochi Hand", cursive',
  fontSize: '1.2em',
})

const Message = ({children}) => (
  <span>
    {isMark(children) && <Span>{children}</Span>}
    {
      R.cond([
        [isO, R.always(` wins!`)],
        [isX, R.always(` wins!`)],
        [R.T, R.always('Its a draw!')]
      ])(children)
    }
  </span>
)

Message.propTypes = {
  children: PropTypes.oneOf([X, O, ''])
}

export default Message

