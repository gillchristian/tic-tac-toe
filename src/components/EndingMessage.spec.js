import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { X, O, s } from '../constants'
import EndingMessage from './EndingMessage'

describe('EndingMessage', () => {
  it('renders ok', () => {
    const wrapper = shallow(<EndingMessage>{X}</EndingMessage>)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a different message for each mark & when empty', () => {
    let wrapper = shallow(<EndingMessage>{O}</EndingMessage>)

    expect(toJson(wrapper)).toMatchSnapshot()

    wrapper = shallow(<EndingMessage>{X}</EndingMessage>)

    expect(toJson(wrapper)).toMatchSnapshot()

    wrapper = shallow(<EndingMessage>{s}</EndingMessage>)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

