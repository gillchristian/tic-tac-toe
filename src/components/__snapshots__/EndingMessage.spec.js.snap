exports[`EndingMessage renders a different message for each mark & when empty 1`] = `
<span>
  <styled.span>
    O
  </styled.span>
   wins!
</span>
`;

exports[`EndingMessage renders a different message for each mark & when empty 2`] = `
<span>
  <styled.span>
    X
  </styled.span>
   wins!
</span>
`;

exports[`EndingMessage renders a different message for each mark & when empty 3`] = `
<span>
  Its a draw!
</span>
`;

exports[`EndingMessage renders ok 1`] = `
<span>
  <styled.span>
    X
  </styled.span>
   wins!
</span>
`;

